graph/src/components/ethereum/types.rs

```diff
@@ -1,12 +1,96 @@
 use ethabi::LogParam;
 use serde::{Deserialize, Serialize};
+use std::cmp::Ordering;
 use web3::types::*;
 
+pub type ThinEthereumBlock = Block<Transaction>;
+
+pub trait ThinEthereumBlockExt {
+    fn number(&self) -> u64;
+    fn transaction_for_log(&self, log: &Log) -> Option<Transaction>;
+    fn transaction_for_call(&self, call: &EthereumCall) -> Option<Transaction>;
+    fn parent_ptr(&self) -> EthereumBlockPointer;
+}
+
+impl ThinEthereumBlockExt for ThinEthereumBlock {
+    fn number(&self) -> u64 {
+        self.number.unwrap().as_u64()
+    }
+
+    fn transaction_for_log(&self, log: &Log) -> Option<Transaction> {
+        log.transaction_hash
+            .and_then(|hash| self.transactions.iter().find(|tx| tx.hash == hash))
+            .cloned()
+    }
+
+    fn transaction_for_call(&self, call: &EthereumCall) -> Option<Transaction> {
+        call.transaction_hash
+            .and_then(|hash| self.transactions.iter().find(|tx| tx.hash == hash))
+            .cloned()
+    }
+
+    fn parent_ptr(&self) -> EthereumBlockPointer {
+        EthereumBlockPointer {
+            hash: self.parent_hash,
+            number: self.number() - 1,
```

Comment:
This needs to check for an overflow: the parent of the genesis block with number `0` is `None`. #1297 turns the result of this to be `Option<EthereumBlockPointer>`.

Reply:
I'm ok if we don't do that in this PR but leave it to #1297 to rebase and add this in.

Reply:
Yes lets fix this when rebasing, I don't mind which PR gets rebased on which.